build/cpp/kvstore/src/kvstore_server.cc

```diff
@@ -3,15 +3,19 @@
 // key value store server constructor
 kvstore_server::KeyValueStoreServiceImpl::KeyValueStoreServiceImpl(
     const std::string &store) {
-  LOG(INFO) << "Server constructor commenced";
-  kvstore_ = new kvstore::KVStore(store);
+  store_ = store;
+  if (store_ != "") {
+    LOG(INFO) << "Reading from store: " << store_;
+  }
+  kvstore_.ReadFile(store_);
 }
 
 // key value store destructor
 kvstore_server::KeyValueStoreServiceImpl::~KeyValueStoreServiceImpl() {
-  LOG(INFO) << "Server destructor commenced";
-  kvstore_->WriteFile();
-  delete kvstore_;
+  if (store_ != "") {
+    LOG(INFO) << "Writing to store: " << store_;
+  }
+  kvstore_.WriteFile(store_);
 }
```

Comment:
No need to unconditionally initialize `store_ = store` 
I suggest all of that code should be encapsulated in the if block to minimize unnecessary branching (ie, calling `ReadFile` and `WriteFile` on empty string if the store flag is empty)

Reply:
This has also been updated.